packages/q/qt5base/xmake.lua

```diff
@@ -0,0 +1,220 @@
+            elseif package:is_arch("x86", "i386") then
+                winarch = "32"
+            else
+                raise("unhandled arch " .. package:targetarch())
+            end
+
+            local compilerVersion
+            if package:is_plat("windows") then
+                local vs = import("core.tool.toolchain").load("msvc"):config("vs")
+                if tonumber(vs) >= 2019 then
+                    compilerVersion = "msvc2019"
+                elseif vs == "2017" or vs == "2015" then
+                    compilerVersion = "msvc" .. vs
+                else
+                    raise("unhandled msvc version " .. vs)
+                end
+
+                if package:is_arch("x64", "x86_64") then
+                    compilerVersion = compilerVersion .. "_64"
+                end
+            else
+                local cc = package:tool("cc")
+                local version = os.iorunv(cc, {"-dumpversion"}):trim()
+
+                import("core.base.semver")
+                local mingw_version = semver.new(version)
+
+                if mingw_version:ge("8.1") then
+                    compilerVersion = "mingw81"
+                elseif mingw_version:ge("7.3") then
+                    compilerVersion = "mingw73"
+                elseif mingw_version:ge("5.3") then
+                    compilerVersion = "mingw53"
+                else
+                    raise("unhandled mingw version " .. version)
+                end
+            end
+
+            arch = "win" .. winarch .. "_" .. compilerVersion
+        elseif package:is_plat("linux") then
+            arch = "gcc_64"
+        elseif package:is_plat("macosx") then
+            arch = "clang_64"
+        elseif package:is_plat("android") then
+            if package:version():le("5.13") then
+                if package:is_arch("x86_64", "x64") then
+                    arch = "android_x86_64"
+                elseif package:is_arch("arm64", "arm64-v8a") then
+                    arch = "android_arm64_v8a"
+                elseif package:is_arch("armv7", "armv7-a") then
+                    arch = "android_armv7"
+                elseif package:is_arch("x86") then
+                    arch = "android_x86"
+                end
+            else
+                arch = "android"
+            end
+        end
+
+        local installdir = package:installdir()
+        os.vrunv("aqt", {"install-qt", "-O", installdir, host, target, version:shortstr(), arch})
+
+        -- move files to root
+        local subdirs = {}
+        if package:is_plat("linux") then
+            table.insert(subdirs, package:is_arch("x86_64") and "gcc_64" or "gcc_32")
+            table.insert(subdirs, package:is_arch("x86_64") and "clang_64" or "clang_32")
+        elseif package:is_plat("macosx") then
+            table.insert(subdirs, package:is_arch("x86_64") and "clang_64" or "clang_32")
+        elseif package:is_plat("windows") then
+            local vs = config.get("vs")
```

Comment:
local vs = import("core.tool.toolchain").load("msvc"):config("vs")

maybe we should move `import("core.tool.toolchain")` to the first line.